appscale/tools/appscale_stats.py

```diff
@@ -0,0 +1,655 @@
+
+  process_stats_headers = [
+    "SERVICE (ID)", "INSTANCES", "UNIQUE MEMORY SUM (MB)",
+    "CPU SUM (%)", "CPU PER 1 PROCESS (%)", "CPU PER 1 CORE (%)"
+  ]
+
+  for proc in raw_process_stats.itervalues():
+    process_list += proc["processes_stats"]
+
+  for proc in process_list:
+    name = proc["unified_service_name"]
+    app_id = proc["application_id"]
+    name_id = name + (" ({})".format(app_id) if app_id else "")
+    if name_id in unique_processes:  # if calculating with process has been made
+      continue
+
+    process_group = [
+      p for p in process_list
+      if p["unified_service_name"] == name
+      and p["application_id"] == app_id
+    ]
+
+    unique_processes.add(name_id)
+    unique_memory = 0
+    cpu_percent = 0
+
+    for p in process_group:
+      unique_memory += int(p["memory"]["unique"])
+      cpu_percent += float(p["cpu"]["percent"])
+      if p["children_num"] > 0:
+        child = p["children_stats_sum"]
+        unique_memory += int(child["memory"]["unique"])
+        cpu_percent += float(child["cpu"]["percent"])
+
+    avg_cpu_per_process = cpu_percent / len(process_group)
+    cpu_percent_cpu_count = cpu_percent / cpu_count
+
+    process_stats.append([
+      name_id,
+      len(process_group),
+      unique_memory / 1024 / 1024,  # row[2] from B to MB
+      cpu_percent,
+      avg_cpu_per_process,
+      cpu_percent_cpu_count
+    ])
+
+  return process_stats_headers, process_stats
+
+
+def get_proxy_stats(raw_proxy_stats, verbose=False, apps_filter=False):
+  """
+  Obtains useful information from proxy statistics and returns:
+  SERVICE (ID), UNIQUE MEMORY SUM (MB), CPU PER 1 PROCESS (%),
+  CPU PER 1 CORE (%), CPU SUM (%) values.
+
+  Args:
+    raw_proxy_stats: A dist in which each key is an ip and value is a dict
+      of useful information.
+    verbose: A boolean - verbose or not verbose mode.
+    apps_filter: A boolean - show all services or applications only.
+
+  Returns:
+    A list of proxy statistics headers.
+    A list of proxy statistics.
+  """
+  proxy_stats_headers = [
+    "SERVICE | ID", "SERVERS", "REQ RATE / REQ TOTAL", "5xx / 4xx", "QUEUE CUR"
+  ]
+
+  proxy_stats_headers_verbose = [
+    "SERVICE (ID)", "SERVERS", "REQ RATE / REQ TOTAL", "5xx / 4xx",
+    "BYTES IN / BYTES OUT", "SESSION CUR / QUEUE CUR", "QTIME / RTIME"
+  ]
+
+  proxy_stats = []
+  proxy_groups = []
+  unique_proxies = {}
+
+  for node in raw_proxy_stats.itervalues():
+    proxy_groups += node["proxies_stats"]
+
+  for node in proxy_groups:
+    if apps_filter:
+      if "application" != node["unified_service_name"]:
+        continue
+
+    service_name_id = node["unified_service_name"]\
+                      + (" | " + node["application_id"]
+                         if node["application_id"]
+                         else "")
+
+    if not service_name_id in unique_proxies:
+      unique_proxies[service_name_id] = {}
+      unique_proxies[service_name_id]["servers_count"] = set()
```

Comment:
Getting `unique_proxies[service_name_id]` so many times can be expensive. I don't think it'll make show stats command slow.. but it still can be improved like this:
```
proxy_summary = unique_proxies.get(service_name_id)
if not proxy_summary:
  unique_proxies[service_name_id] = proxy_summary = {}
  proxy_summary["servers_count"] = set()
  ...
```